### ISSUE TITLE
Component for representing Events on a timeline

### ISSUE THREAD

**Issue by hernil** [cognitedata/gearbox.js#520]
We suggest a component that can be used to visualize Events from CDF on a timeline. 
![bilde](https://user-images.githubusercontent.com/1758308/69868603-7252f100-12aa-11ea-9cdd-5ee437f5ca63.png)

As seen in the sketch this will often be used together with a `TimeSeriesChart`, but we suggest that it should be a seperate component anyway. Note that the sketch shows three separate timelines. 

## Suggested features and APIs
- Define a time period by start and end Datetimes 
- Pass in an array of Event(id)s that will then be distributed on the timeline
  - One should be able to decide whether one wants to visualize the enitre event duration as a "block" or outline, or if one is only interested in the events' starttime as a descrete happening (the latter case should then have some sort of min width) 
- Decide a timelines color, as well as each event color (desired use case is for grouping of event types, but simply allowing to set custom colors allows for greater flexibility)
- Clicking (or mouseover) an event on the timeline should give a brief summary of the event
  - This summary should probably be customizable so that one can decide what data and metadata to show. Probably best to allow a callback like `(event) => ReactNode`. 

## Challenges and changes required outside of the component
As this component may want to be used in conjunction with a `TimeSeriesChart` or even multiple instanses of itself we will need a way for a parent component to know where the mouse pointer hovers on the timeline of these components so as to reflect this value to sibling components that may then draw an apropriate cursor themselves. The parent component will also need to be made aware of changes in the zoom-level of a `Timeline` or `TimeSeriesChart` component so siblings can reflect those as well. 

## In conclusion 
We hope that the described component, although solving a specific need for us, has been made general enough as to be a valueable addition to the Gearbox library. Do not hesitate to get in touch for more details or clarifications.

**Comment by mrtraser** [comment]
Few things to consider: 
- How should we define number of timelines and decide which event should be placed on each of timeline?

Suggested API (not final):
```typescript tsx
interface TimeLineEvent {
  eventId: number;
  view: "block" | "outline";
  type: "discrete" | "continuous";
  summary: HTMLElement;
}
interface TimeLine {
  name: string;
  events: TimeLineEvent[];
  color: string;
}
const timelines: TimeLine[] = [
  {
    name: "outage",
    events: [
      {
        eventId: 123,
        view: "block",
        type: "discrete",
        summary: <p>More Info</p>
      }
    ],
    color: "red"
  },
  {
    name: "crash",
    events: [
      {
        eventId: 321,
        view: "outline",
        type: "continuous",
        summary: <p>More Info</p>
      }
    ],
    color: "blue"
  }
];
<TimeLineComponent timeLines={timelines} />;
```

- How should we render events which overlaps on a timeline? (TBD)

**Comment by mrtraser** [comment]
Closed by #614
